drivers/display/ssd1322.c

```diff
@@ -0,0 +1,344 @@
+		return -1;
+	}
+
+	cmd = SSD1322_SET_ROW_ADDR;
+	cmd_data[0] = y;
+	cmd_data[1] = (y + desc->height - 1);
+	if (ssd1322_write_bus(dev, &cmd, sizeof(cmd), true)) {
+		LOG_ERR("Failed to write command");
+		return -1;
+	}
+	if (ssd1322_write_bus(dev, cmd_data, sizeof(cmd_data), false)) {
+		LOG_ERR("Failed to write command data");
+		return -1;
+	}
+
+	cmd = SSD1322_ENABLE_RAM_WRITE;
+	if (ssd1322_write_bus(dev, &cmd, sizeof(cmd), true)) {
+		LOG_ERR("Failed to write command");
+		return -1;
+	}
+
+	uint8_t *buff_ptr = (uint8_t *)buf;
+
+	for (uint16_t j = 0; j < buf_len; j++) {
+		uint8_t pixel_buff[4] = {0};
+
+		for (uint8_t i = 0; i < 8; ++i) {
+			if (*buff_ptr & (1 << i)) {
+				uint8_t nibble_pos = (i & 1) ? 0 : 4;
+
+				pixel_buff[i >> 1] |= 0x0F << nibble_pos;
+			}
+		}
+		if (ssd1322_write_bus(dev, pixel_buff, sizeof(pixel_buff), false)) {
+			LOG_ERR("Failed to write pixel data");
+			return -1;
+		}
+		buff_ptr++;
+	}
+
+	return 0;
+}
+
+static int ssd1322_read(const struct device *dev, const uint16_t x,
+			const uint16_t y,
+			const struct display_buffer_descriptor *desc,
+			void *buf)
+{
+	LOG_ERR("Unsupported");
+	return -ENOTSUP;
+}
+
+static void *ssd1322_get_framebuffer(const struct device *dev)
+{
+	LOG_ERR("Unsupported");
+	return NULL;
+}
+
+static int ssd1322_set_brightness(const struct device *dev,
+		const uint8_t brightness)
+{
+	LOG_WRN("Unsupported");
+	return -ENOTSUP;
+}
+
+static int ssd1322_set_contrast(const struct device *dev, const uint8_t contrast)
+{
+	uint8_t cmd = SSD1322_SET_CONTRAST;
+
+	if (ssd1322_write_bus(dev, &cmd, sizeof(cmd), true)) {
+		LOG_ERR("Failed to write command");
+		return -1;
+	}
+	uint8_t data = contrast;
+
+	return ssd1322_write_bus(dev, &data, sizeof(data), false);
+}
+
+static void ssd1322_get_capabilities(const struct device *dev,
+				     struct display_capabilities *caps)
+{
+	const struct ssd1322_config *config = dev->config;
+
+	memset(caps, 0, sizeof(struct display_capabilities));
+	caps->x_resolution = config->width;
+	caps->y_resolution = config->height;
+	caps->supported_pixel_formats = PIXEL_FORMAT_MONO01;
+	caps->current_pixel_format = PIXEL_FORMAT_MONO01;
+}
+
+static int ssd1322_set_orientation(const struct device *dev,
+				   const enum display_orientation orientation)
+{
+	LOG_ERR("Unsupported");
+	return -ENOTSUP;
+}
+
+static int ssd1322_set_pixel_format(const struct device *dev,
+				    const enum display_pixel_format pf)
+{
+	if (pf == PIXEL_FORMAT_MONO01) {
+		return 0;
+	}
+	LOG_ERR("Unsupported");
+	return -ENOTSUP;
+}
+
+static int ssd1322_init_device(const struct device *dev)
+{
+	const struct ssd1322_config *config = dev->config;
+
+	if (gpio_pin_set_dt(&config->reset, 1) < 0) {
+		return -1;
+	}
+	k_sleep(K_MSEC(10));
+	if (gpio_pin_set_dt(&config->reset, 0) < 0) {
+		return -1;
+	}
+	k_sleep(K_MSEC(50));
+
+	int ret = 0;
+	uint8_t cmd, data[2];
+
+	cmd = SSD1322_DISPLAY_OFF;
+	ret |= ssd1322_write_bus(dev, &cmd, sizeof(cmd), true);
```

Comment:
I get the idea here, but I think we should be checking `ret` after each call, and returning if there is an error. This implementation will be more challenging to debug, since you won't know *where* something failed without stepping through with a debugger and watching the value of `ret`

Reply:
Done.